subprojects/plugins/src/main/java/org/gradle/api/component/internal/DefaultConfigurationBackedConsumableVariant.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.component.internal;
+
+import org.gradle.api.Action;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.DependencyConstraint;
+import org.gradle.api.attributes.AttributeContainer;
+import org.gradle.api.capabilities.CapabilitiesMetadata;
+import org.gradle.api.component.ConfigurationBackedConsumableVariant;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.attributes.AttributeContainerInternal;
+import org.gradle.internal.component.external.model.ImmutableCapabilities;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Default implementation of {@link ConfigurationBackedConsumableVariant}.
+ */
+public class DefaultConfigurationBackedConsumableVariant implements ConfigurationBackedConsumableVariant {
```

Comment:
💭 Should this either be `abstract` and always created via `ObjectFactory`, or drop the `@Inject` and always be created with `new` (even by factories), for consistency?